pkg/controllers/ipam/ipam.go

```diff
@@ -176,24 +329,50 @@ func (i *IPAM) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	clone := eip.DeepCopy()
 
 	if err = i.updateEip(clone); err != nil {
-		i.Client.Status().Update(context.Background(), clone)
+		i.Client.Status().Update(ctx, clone)
 		i.Event(eip, v1.EventTypeWarning, EipAddOrUpdateReason, fmt.Sprintf("%s: %s", util.GetNodeName(), err.Error()))
 		return ctrl.Result{}, err
 	}
 
 	if reflect.DeepEqual(clone.Status, eip.Status) {
 		return ctrl.Result{}, nil
 	}
-	i.updateMetrics(eip)
+	//i.updateMetrics(eip)
 	return ctrl.Result{}, i.Client.Status().Update(context.Background(), clone)
 }
 
-func (i *IPAM) updateEip(e *networkv1alpha2.Eip) error {
-	var (
-		sp  speaker.Speaker
-		err error
-	)
+func (i *IPAM) AssignIP(request *IPAMRequest) (result IPAMResult, err error) {
+	eip := &networkv1alpha2.Eip{}
+	err = i.Get(context.Background(), types.NamespacedName{Name: request.Allocate.Eip}, eip)
+	if err != nil {
+		return result, err
+	}
 
+	clone := eip.DeepCopy()
+	if !eip.Status.Ready {
+		return result, fmt.Errorf("layer2 eip speaker not ready")
+	}
+
+	addr := request.assignIPFromEip(clone)
+	if addr == "" {
+		return result, fmt.Errorf("no avliable eip")
```

Comment:
Just a question, Should we record  some events for this situation?

Reply:
> Just a question, Should we record some events for this situation?

yes, I think we can add related events where necessary, and we can improve this problem alone.